src/test/java/org/javarosa/core/model/instance/geojson/GeoJsonExternalInstanceTest.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2022 ODK
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.javarosa.core.model.instance.geojson;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.fail;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import org.javarosa.core.model.instance.TreeElement;
+import org.junit.Test;
+
+public class GeoJsonExternalInstanceTest {
+    @Test
+    public void parse_throwsException_ifTopLevelElementNotFeatureCollection() {
+        try {
+            GeoJsonExternalInstance.parse("id", new ByteArrayInputStream(GEOMETRY_COLLECTION_GEOJSON.getBytes(StandardCharsets.UTF_8)));
+            fail("Exception expected");
+        } catch (IOException e) {
+            // expected
+        }
+    }
+
+    @Test
+    public void parse_parsesMultipleFeatures() throws IOException {
```

Comment:
The breakup of these tests is very TDD. I like it. It might feel simpler now you've written them to combine `parse_parsesMultipleFeatures` and `parse_addsGeometryAsChild` - that'll give you a kind of "core behaviour" test. I'll leave that up to you though!